src/main/java/io/github/syst3ms/skriptparser/expressions/CondExprEmpty.java

```diff
@@ -0,0 +1,63 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import java.util.Arrays;
+
+import org.jetbrains.annotations.Nullable;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.base.ConditionalExpression;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+
+/**
+ * Check if a given string or array is empty.
+ *
+ * @name Empty
+ * @pattern %strings/objects% (is|are)[1:( not|n't)] empty
+ * @since ALPHA
+ * @author Olyno
+ */
+public class CondExprEmpty extends ConditionalExpression {
+
+    static {
+        Parser.getMainRegistration().addExpression(CondExprEmpty.class,
+            Boolean.class,
+            true,
+            2,
+            "%strings/objects% (is|are)[1:( not|n't)] empty"
+        );
+    }
+
+    private Expression<?> expr;
+
+    @Override
+    public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+        expr = (Expression<?>) expressions[0];
+        setNegated(parseContext.getParseMark() == 1);
+        return true;
+    }
+
+    @Override
+    public boolean check(TriggerContext ctx) {
+        Object[] values = expr.getValues(ctx);
+        if (values.length == 0)
+            return true != isNegated();
+        if (values.length == 1)
+            return (values[0] instanceof String == false && values[0] == null) != isNegated();
```

Comment:
I don’t like doing an ‘equals’-operator on a boolean. Just use the ‘!’-operator for this.
Also values[0] can never be null, so it’s redundant to check for that.